src/utils/dev.ts

```diff
@@ -147,6 +147,21 @@ class NuxtDevServer extends EventEmitter {
       },
     })
 
+    // Connect Vite HMR
+    this._currentNuxt.hooks.hookOnce(
+      'vite:extendConfig',
+      (config, { isClient }) => {
+        if (isClient && config.server) {
+          config.server.hmr = {
+            server: this.listener.server,
+          }
```

Comment:
is it worth merging configs in case user has set `overlay` or `timeout` or otherwise configured this?

Reply:
Mainly i wanted to avoid vite confliciting with (nuxt provided) port options. Let me try.

Reply:
We can update nuxt to disallow setting certain options in `vite.server.hmr` if we want to take full control of them. of course in most cases users will not configure this but there may be some edge cases

Reply:
we only nullify intentional keys now: cad706b6696f1c335da266e57d65f91dba1d4d5a

Surely would be nice to also limit in types if we finally want to support main port 👍🏼